rclpy/rclpy/qos.py

```diff
@@ -471,3 +471,21 @@ def short_keys(cls):
     def get_from_short_key(cls, name):
         """Retrieve a policy type from a short name, case-insensitive."""
         return cls[name.upper()].value
+
+
+class QoSCompatibility(IntEnum):
+    OK = 0
+    WARNING = 1
+    ERROR = 2
+
+
+def qos_check_compatible(publisher_qos: QoSProfile, subscription_qos: QoSProfile):
```

Comment:
It would be good to add a docstring for this function, similar to the C++ equivalent.